internal/jennies/java/converter.go

```diff
@@ -0,0 +1,167 @@
+package java
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/grafana/codejen"
+	"github.com/grafana/cog/internal/ast"
+	"github.com/grafana/cog/internal/jennies/common"
+	"github.com/grafana/cog/internal/jennies/template"
+	"github.com/grafana/cog/internal/languages"
+	"github.com/grafana/cog/internal/tools"
+)
+
+type Converter struct {
+	config         Config
+	nullableConfig languages.NullableConfig
+	tmpl           *template.Template
+}
+
+func (jenny *Converter) JennyName() string {
+	return "JavaConverter"
+}
+
+func (jenny *Converter) Generate(context languages.Context) (codejen.Files, error) {
+	files := codejen.Files{}
+
+	for _, builder := range context.Builders {
+		output, err := jenny.generateConverter(context, builder)
+		if err != nil {
+			return nil, err
+		}
+
+		filename := filepath.Join(
+			jenny.config.ProjectPath,
+			formatPackageName(builder.Package),
+			fmt.Sprintf("%sConverter.java", tools.UpperCamelCase(builder.Name)),
+		)
+
+		files = append(files, *codejen.NewFile(filename, output, jenny))
+	}
+
+	var err error
+	for _, schema := range context.Schemas {
+		schema.Objects = schema.Objects.Filter(func(key string, obj ast.Object) bool {
+			if obj.Type.ImplementedVariant() != string(ast.SchemaVariantDataQuery) {
+				return false
+			}
+
+			return !obj.Type.HasHint(ast.HintSkipVariantPluginRegistration)
+		})
+
+		schema.Objects.Iterate(func(key string, obj ast.Object) {
+			output, genErr := jenny.generateDataqueryConverter(context, schema, obj)
+			if genErr != nil {
+				err = genErr
+			} else {
+				filename := filepath.Join(
+					jenny.config.ProjectPath,
+					formatPackageName(schema.Package),
+					fmt.Sprintf("%sMapperConverter.java", tools.UpperCamelCase(obj.Name)),
+				)
+
+				files = append(files, *codejen.NewFile(filename, output, jenny))
+			}
+		})
+	}
+
+	return files, err
+}
+
+func (jenny *Converter) generateConverter(context languages.Context, builder ast.Builder) ([]byte, error) {
+	converter := languages.NewConverterGenerator(jenny.nullableConfig).FromBuilder(context, builder)
+
+	schema, schemaFound := context.Schemas.Locate(builder.Package)
+	isPanel := schemaFound && schema.Metadata.Variant == ast.SchemaVariantPanel && builder.Name == "Panel"
+
+	imports := NewImportMap(jenny.config.PackagePath)
+	packageMapper := func(pkg string, class string) string {
+		if imports.IsIdentical(pkg, schema.Package) {
+			return ""
+		}
+
+		return imports.Add(class, pkg)
+	}
+	typeFormatter := createFormatter(context, jenny.config).withPackageMapper(packageMapper)
+
+	builderNameFormat := fmt.Sprintf("%s.Builder", tools.UpperCamelCase(converter.BuilderName))
```

Comment:
There are some builders that the name of the class is different than the builder because they are renamed by a veneer. Apart of this, we have classes with different constructors depending of they type (`VariableModel` for example), and the builder name is also different.